### PR TITLE
Detection rule new detection ux

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
Replaced the detection code editor in Detection rule creation form to use UI components. It simplifies entering detection and also allows user to enter values for a field by uploading text or csv files.

image

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+                              })
+                            }
+                            value={datum.field}
+                          />
+                        </EuiFormRow>
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false} style={{ minWidth: 200 }}>
+                        <EuiFormRow label={<EuiText size={'s'}>Modifier</EuiText>}>
+                          <EuiComboBox
+                            // isInvalid={isInvalidInputForQuery('logType')}
```
Comment (AWSHurneyt) [inline public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx:494]: Is this dev code that can be removed?

File: public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+                              values,
+                            });
+                          }}
+                          value={datum.values.join('\n')}
+                          compressed={true}
+                        />
+                      </>
+                    ) : (
+                      <EuiFieldText
+                        // isInvalid={props.touched.name && !!props.errors.name}
```
Comment (AWSHurneyt) [inline public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx:558]: Dev code?

File: public/utils/validation.ts [hunk 3]
```diff
@@ -10,6 +10,12 @@ export const MAX_NAME_CHARACTERS = 50;
 // numbers 0-9, hyphens, spaces, and underscores.
 export const NAME_REGEX = new RegExp(/^[a-zA-Z0-9 _-]{5,50}$/);
 
+export const DETECTION_NAME_REGEX = new RegExp(/^[a-zA-Z0-9_]{5,50}$/);
+
+export const CONDITION_REGEX = new RegExp(
```
Comment (AWSHurneyt) [inline public/utils/validation.ts:15]: Nitpick: Similar to what you have on line 10, it could be helpful to include an explanation of the regex here.

**Review by AWSHurneyt — APPROVED** [review]

File: public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx (hunk 2 above)
Comment (amsiglan) [reply public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx]: added validation

File: public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx (hunk 1 above)
Comment (amsiglan) [reply public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx]: added validation in latest iteration

File: public/utils/validation.ts (hunk 3 above)
Comment (amsiglan) [reply public/utils/validation.ts:17]: added

**Review by eirsep — APPROVED** [review]

**Pull request merged**